pennylane_lightning/core/_serialize.py

```diff
@@ -183,19 +184,21 @@ def _named_obs(self, observable, wires_map: dict = None):
 
     def _hermitian_ob(self, observable, wires_map: dict = None):
         """Serializes a Hermitian observable"""
-        assert not isinstance(observable, Tensor)
+        assert not isinstance(observable, (Tensor, Prod))
```

Comment:
Yeah I don't see the point of having this assertion.

> What about other obs like Hamiltonian, SProd, and Sum?

If we want to keep the assertion it makes sense to add `Hamiltonian` to the list, but `Sum` and `SProd` are handled by this branch. I think we should update `QuantumScriptSerializer._ob` to dispatch to `_hamiltonian_ob` for `SProd` and `Sum` though. What are your thoughts? cc @vincentmr @AmintorDusko

Reply:
I agree, I think we can remove that assertion. 

Since it's almost Friday, I'd say if a change doesn't work off the bat, let's follow-up in Q2. For `SProd` and `Sum`, can't the underlying observables be anything, including Hermitian observables? In this case, we'd need more checks to dispatch to `_hamiltonian` no?

Reply:
I don't think we should. The recursive structure of the serialization should accommodate for this. We might just end up with weird looking serializations, but they will still be correct.

Reply:
Sounds good to me! Let's focus on the Q1 deliverables.  But please create follow-up stories for Q2.

Reply:
If there is any point about keeping this assertion, I need to hear it as I am not yet convinced :)

Reply:
I'm convinced, I don't think we need the assertion. I will remove it.